python/ray/tests/test_ray_init.py

```diff
@@ -366,6 +368,52 @@ def test_ray_init_using_hostname(ray_start_cluster):
     assert node_table[0].get("NodeManagerHostname", "") == hostname
 
 
+def test_hosted_external_dashboard_url(shutdown_only):
+    """
+    Test setting external dashboard URL through environment variable.
+    """
+    orig_external_dashboard_url = os.environ.get(RAY_OVERRIDE_DASHBOARD_URL)
+
+    os.environ[RAY_OVERRIDE_DASHBOARD_URL] = "https://external_dashboard_url"
+    info = ray.init()
+    assert info.dashboard_url == "external_dashboard_url"
+    assert info.address_info["webui_url"] == "external_dashboard_url"
+    assert (
+        info.address_info["webui_url_with_protocol"] == "https://external_dashboard_url"
+    )
+    assert (
+        ray_address_to_api_server_url("auto")
+        == info.address_info["webui_url_with_protocol"]
+    )
+    ray.shutdown()
+
+    os.environ[RAY_OVERRIDE_DASHBOARD_URL] = "external_dashboard_url"
+    info = ray.init()
+    assert info.dashboard_url == "external_dashboard_url"
+    assert info.address_info["webui_url"] == "external_dashboard_url"
+    assert (
+        info.address_info["webui_url_with_protocol"] == "http://external_dashboard_url"
```

Comment:
can you also test the case when a given external URL is invalid?

Reply:
The `urlparse` method should work on invalid urls even if there is no protocol (eg: `external_dashboard_url`) or if the protocol is invalid